Add perimeter_ext function to LineMeasurableExt trait

zhangfengcdt opened the pull request:
Implements a Perimeter trait for the geo-generic-alg crate that calculates the perimeter of geometries using generic WKB traits, following the same pattern as the existing Area trait.

- Follows the same generic WKB transformation pattern as area.rs
- Uses CoordFloat bounds for floating-point calculations with hypot() method
- Comprehensive test coverage matching area.rs patterns

This provides a cleaner API where users can call .perimeter() directly on polygons instead of manually summing exterior and interior ring lengths.

Test Coverage
- All geometry types with various configurations
- Edge cases (empty geometries, single points, degenerate cases)
- Numerical stability testing with large coordinate shifts
- Complex polygons with manual calculation verification
- WKT parsing integration
- Mixed geometry collections

zhangfengcdt committed:
Implement perimeter based on Generic Wkb traits

zhangfengcdt committed:
add benchmarks

zhangfengcdt committed:
fix fmt lint

zhangfengcdt committed:
address pr comments

Kontinuation commented on geo-generic-alg/src/algorithm/perimeter.rs:
```diff
@@ -0,0 +1,624 @@
+        let start_coord = line.start_coord();
+        let end_coord = line.end_coord();
+        let delta = end_coord - start_coord;
+        perimeter = perimeter + delta.x.hypot(delta.y);
```
We have implemented `perimeter_trait` for `LineTraitExt`. Why not directly call `line.perimeter_trait()`?

Kontinuation commented on geo-generic-alg/src/algorithm/perimeter.rs:
```diff
@@ -0,0 +1,624 @@
+        let start_coord = self.start_coord();
+        let end_coord = self.end_coord();
+        let delta = end_coord - start_coord;
+        delta.x.hypot(delta.y)
```
We can implement perimeter based on `LengthMeasurableExt`, which is a cleaner and more flexible approach, and also better fit with the overall design of geo line measurement algorithms.

Kontinuation commented on geo-generic-alg/src/algorithm/perimeter.rs:
```diff
@@ -0,0 +1,624 @@
+    }
+}
+
+// Critical: GeometryTag implementation for WKB compatibility
```
This comment looks weird. There's nothing specific about WKB in this trait.

Kontinuation reviewed (approved):
I can bear with the chaos of our new line measurement related algorithms not being aligned with the original design of geo, but we may come back to rework this some day.

zhangfengcdt committed:
Use alternative implementation based on LengthMeasurableExt

and remove the Perimeter trait

zhangfengcdt committed:
fix tests and benchmarks

zhangfengcdt commented:
> I can bear with the chaos of our new line measurement related algorithms not being aligned with the original design of geo, but we may come back to rework this some day.

Yes, thanks for the suggestion. It is actually much cleaner to add perimeter to LineMeasurableExt trait. I added a new perimeter_ext because I think the callers (e.g., st_length, st_perimeter) will still need to differentiate them. Also removed the Perimeter trait completely.
